src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Completion/TagHelperCompletionProvider.cs

```diff
@@ -119,20 +119,16 @@ public override IReadOnlyList<RazorCompletionItem> GetCompletionItems(RazorCompl
                 var attributeCompletions = GetAttributeCompletions(parent, containingTagNameToken.Content, selectedAttributeName, stringifiedAttributes, context.TagHelperDocumentContext, context.Options);
                 return attributeCompletions;
 
-                static bool IsPossiblePartiallyWrittenAttribute(SyntaxNode? attributeSyntax, int absoluteIndex)
+                static bool InOrAtEndOfAttribute(SyntaxNode attributeSyntax, int absoluteIndex)
                 {
                     // When we are in the middle of writing an attribute it is treated as a minimilized one, e.g.:
                     // <form asp$$ - 'asp' is parsed as MarkupMinimizedTagHelperAttributeSyntax (tag helper)
                     // <SurveyPrompt Tit$$ - 'Tit' is parsed as MarkupMinimizedTagHelperAttributeSyntax as well (razor component)
                     // Need to check for MarkupMinimizedAttributeBlockSyntax in order to handle cases when html tag becomes a tag helper only with certain attributes
-                    // We allow the absoluteIndex to be anywhere in the attribute, so that `<SurveyPrompt T$$it` doesn't return only
-                    // the html completions, because that has the effect of overwriting the casing of the attribute.
-                    if (attributeSyntax is MarkupMinimizedTagHelperAttributeSyntax or MarkupMinimizedAttributeBlockSyntax)
-                    {
-                        return attributeSyntax.Span.Start < absoluteIndex && attributeSyntax.Span.End >= absoluteIndex;
-                    }
-
-                    return false;
+                    // We allow the absoluteIndex to be anywhere in the attribute, and for non minimized attributes,
+                    // so that `<SurveyPrompt Title=""` doesn't return only the html completions, because that has the effect of overwriting the casing of the attribute.
+                    return attributeSyntax is MarkupMinimizedTagHelperAttributeSyntax or MarkupMinimizedAttributeBlockSyntax or MarkupTagHelperAttributeSyntax &&
```

Comment:
What about non-tag helper full attributes? See my comment above: if a tag helper has restriction, so it becomes a tag helper only with certain attributes, will it work as expected in this case?

Reply:
Your "class" case? I got no completions for that, as expected